app/Http/Controllers/JobSummaryController.php

```diff
@@ -62,39 +64,49 @@ public function show(Request $request, JobPoster $job)
             'disabled' => !$job->isClosed(),
         ];
 
-        switch ($job->job_status_id) {
-            case 1:
-                $status = Lang::get('common/lookup/job_status.draft');
-                break;
-            case 2:
-                $status = Lang::get('common/lookup/job_status.in_review');
-                break;
-            case 3:
-                $status = Lang::get('common/lookup/job_status.approved');
-                break;
-            case 4:
-                $status = Lang::get('common/lookup/job_status.open');
-                break;
-            case 5:
-                $status = Lang::get('common/lookup/job_status.closed');
-                break;
-            case 6:
-                $status = Lang::get('common/lookup/job_status.complete');
-                break;
-        }
-        // TODO: This should change based on the current status.
-        $status_description = $job->job_status_id == 2
-            ? $summaryLang['under_review']
-            : '';
+        $status = $job->job_poster_status->name;
+        $status_description = $job->job_poster_status->description;
 
         $portal = '';
-
         if (WhichPortal::isHrPortal()) {
             $portal = 'hr';
         } elseif (WhichPortal::isManagerPortal()) {
             $portal = 'manager';
         }
 
+        $transitionManager = new JobStatusTransitionManager();
+        $transitionToButton = function (JobPosterStatusTransition $transition) use ($user, $job, $transitionManager) {
```

Comment:
Not suggesting you change this, but this is a very Javascript way of writing PHP, and something Scrutinizer wouldn't be a fan of.

Traditionally you would do something like
```php
foreach ($transitionManager->legalTransitions($job->job_poster_status->key)) {
    array_push($buttonGroups, $this->transitionToButton($transition, $user, $job, $transitionManager));
}
```

With the `transitionToButton` function being defined outside of the `show` method here.

Again, not saying you need to refactor, and I know it's a byproduct of hopping between languages frequently, just something to keep in mind.

Reply:
Thanks. This is less a byproduct of hopping between languages, and more about me being less familiar with PHP conventions in the first place. Most of my reading is about javascript, or even more functional languages.